tools/tpm2_tool.c

```diff
@@ -61,18 +62,101 @@ static ESYS_CONTEXT* ctx_init(TSS2_TCTI_CONTEXT *tcti_ctx) {
     return esys_ctx;
 }
 
+/*
+ * Build a list of the TPM2 tools linked into this executable
+ */
+static const tpm2_tool ** tools;
+static unsigned tool_count;
+
+void tpm2_tool_register(const tpm2_tool * tool)
+{
+    if (!tools)
+        tools = calloc(1024, sizeof(*tools));
+
+    //printf("%s: register %s\n", __func__, tool->name);
+    if (tools)
+        tools[tool_count++] = tool;
+}
+
+static const char * tpm2_tool_name(const char * arg)
+{
+    const char * name = rindex(arg, '/');
+    if (name)
+        name++; // skip the '/'
+    else
+        name = arg; // use the full executable name as is
+
+    if (strncmp(name, "tpm2_", 5) == 0)
+        name += 5;
+
+    return name;
+}
+
+static const tpm2_tool * tpm2_tool_lookup(int *argc, char *** argv)
+{
+    // no tools linked in?
+    if (!tools)
+        return NULL;
+
+    // only one tool linked in? use it no matter the program name.
+    if (tool_count == 1)
```

Comment:
no, if you run tpm2 just die with  expected commandlet name or something. Consistent with other commandlets like git.

Reply:
Looks like return NULL.

Reply:
That's left over from when I was still building the tools as separate programs in addition to the bundled one.  That way if you copied `tpm2_pcrread` to `foo`, it would still read the PCRs since that tool had only a single handler in the array.  It can go away now.